test/gtest/common/test_helpers.cc

```diff
@@ -895,4 +895,18 @@ void skip_on_address_sanitizer()
 #endif
 }
 
+bool running_inside_valgrind()
```

Comment:
why just RUNNING_ON_VALGRIND is not enough?

Reply:
We run tests configured for valgrind twice, once without and once with valgrind wrapper. For the first occurence, the yield is actually killing performance expectations. Saw it on CI `ud/test_ucp_perf.envelope/37` that seems to have started failing.

Reply:
I'd expect RUNNING_ON_VALGRIND to be 0 when the tests are configured with valgrind but running without valgrind

Reply:
actually you're right, reverted that part and pushed, let's see CI, most likely the failure was a coincidence.